certbot/tests/compat/filesystem_test.py

```diff
@@ -6,41 +6,162 @@
 from certbot.tests.util import TempDirTestCase
 
 
-class FilesystemTest(TempDirTestCase):
-    """Unit tests for filesystem module."""
-    @unittest.skipIf(os.name != 'nt', reason='Test specific to Windows security')
+@unittest.skipIf(os.name != 'nt', reason='Test specific to Windows security')
+class WindowsChmodTests(TempDirTestCase):
+    """Unit tests for Windows chmod function in filesystem module"""
+    def test_symlink_resolution(self):
+        probe_path = _create_probe(self.tempdir)  # This is 0o744 by default
+
+        link_path = os.path.join(self.tempdir, 'link')
+        os.symlink(probe_path, link_path)
+
+        ref_dacl_probe = _get_security_dacl(probe_path).GetSecurityDescriptorDacl()
+        ref_dacl_link = _get_security_dacl(link_path).GetSecurityDescriptorDacl()
+
+        # Removing the rights for `all`, we have at least one ACL less than in the case of 0o744.
+        filesystem.chmod(link_path, 0o700)
+
+        # Assert the real file is impacted, not the link.
+        cur_dacl_probe = _get_security_dacl(probe_path).GetSecurityDescriptorDacl()
+        cur_dacl_link = _get_security_dacl(link_path).GetSecurityDescriptorDacl()
+        self.assertFalse(filesystem._compare_dacls(ref_dacl_probe, cur_dacl_probe))  # pylint: disable=protected-access
+        self.assertTrue(filesystem._compare_dacls(ref_dacl_link, cur_dacl_link))  # pylint: disable=protected-access
```

Comment:
I have mixed feelings about using this private, production code in our tests, but it's probably fine. As a nit, what do you think about creating a `_compare_dacls` function in this file and just disabling `pylint` there?

Reply:
In fact, I imported `_compare_dacls` from a future PR, as it is useful here. And this function is normally in `filesystem`, as it is used to check modes.

Reply:
I understand all that, but what do you think about my nit?

Reply:
I think it is not necessary here, because it is only some tricks to compare objects. I can make it public, and add a specific unit test for it if you want.

Reply:
Well, after the huge simplification done to this function, I think we can pay the price of a 2 LOC duplicated in favor of test readability.